### ISSUE TITLE
Service Remains at 0/1 Replicas if Bind-mount Requested

### ISSUE THREAD

**Issue by Kent-H** [moby/moby#32630]
**Description**

When using swarm mode, bind-mounting a volume or file to a service causes that service to permanently remain at zero replicas.

**Steps to reproduce the issue:**
1. `$ docker service create --name mount --mount 'type=bind,source=/e/Username/folder-that-exists/,destination=/path/in/container/' ubuntu:14.04 ping google.ca`
`jqo9fllnhouqvjd19e77jkpkk`
2. `$ docker service create --name no-mount ubuntu:14.04 ping google.ca`
`xv6r2k7s3j6er01o8z1is48ve`

**Describe the results you received:**
```
$ docker service ls
ID            NAME      MODE        REPLICAS  IMAGE
jqo9fllnhouq  mount     replicated  0/1       ubuntu:14.04
xv6r2k7s3j6e  no-mount  replicated  1/1       ubuntu:14.04
```

**Describe the results you expected:**

The services should have one replica each, but the bind-mounted one will never start.

**Additional information you deem important (e.g. issue happens only occasionally):**

Currently running a single docker instance in swarm mode.  I have tried with both unix-style and windows-style directories (`/e/directories` & `E:\directories`)

Containers with mounted volumes run fine if started via `docker run`.

**Output of `docker version`:**

```
Client:
 Version:      17.03.1-ce
 API version:  1.27
 Go version:   go1.7.5
 Git commit:   c6d412e
 Built:        Tue Mar 28 00:40:02 2017
 OS/Arch:      windows/amd64

Server:
 Version:      17.03.1-ce
 API version:  1.27 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   c6d412e
 Built:        Fri Mar 24 00:00:50 2017
 OS/Arch:      linux/amd64
 Experimental: true
```

**Output of `docker info`:**

```
Containers: 1
 Running: 1
 Paused: 0
 Stopped: 0
Images: 17
Server Version: 17.03.1-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
Swarm: active
 NodeID: 2i9dc5vkjlsybo37rl6e4nutc
 Is Manager: true
 ClusterID: aq4pnj9q6nimmabvdq8c5rwv0
 Managers: 1
 Nodes: 1
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
 Node Address: 192.168.65.2
 Manager Addresses:
  192.168.65.2:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 4ab9917febca54791c5f071a9d1f404867857fcc
runc version: 54296cf40ad8143b62dbcaa1d90e520a2136ddfe
init version: N/A (expected: 949e6facb77383876aeff8a6944dde66b3089574)
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.13-moby
Operating System: Alpine Linux v3.5
OSType: linux
Architecture: x86_64
CPUs: 7
Total Memory: 1.934 GiB
Name: moby
ID: RWPI:VOZW:P45I:UJOT:KP5E:DVSS:FLKN:AQPG:DUBF:OA7D:RSFO:QDR2
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 35
 Goroutines: 136
 System Time: 2017-04-14T17:41:41.7405516Z
 EventsListeners: 1
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Physical Windows 10.  Nothing super special.

**Comment by thaJeztah** [comment]
ping @johnstep @friism

**Comment by friism** [comment]
Hm, this is a fun one. @Kent-H you're using Docker for Windows, right? If that's the case, we have actually fixed this, but you probably have to be on Edge to get it. Can you try installing the Edge version of Docker for Windows?

Internal issue is 6694 for anyone reading along. cc @simonferquel

**Comment by Kent-H** [comment]
Using the latest edge version appears to fix the problem.  Thanks for the tip!

**Comment by Kent-H** [comment]
Ubuntu's edge version appears to be behind the windows version.

"Docker for Windows" edge version is `17.05.0`, everything works as expected, no issues.
Under Ubuntu, the only edge version available is `17.04.0`, where this is still broken.

**Comment by thaJeztah** [comment]
@Kent-H the "edge" versions of Docker for Windows / Docker for Mac currently includes release candidates. Docker 17.05 has not yet been released (but release candidates are available), which is where the difference in version comes from

**Comment by thaJeztah** [comment]
Given that the issue is resolved in 17.05, I'll close the issue, but feel free to continue the conversation
